bindings/java/src/main/java/org/ucx/jucx/NativeLibs.java

```diff
@@ -99,13 +100,20 @@ private static void extractUCTLibs() {
         }
 
         uctLibs.forEach(filePath -> {
+            if (!filePath.getFileName().toString().endsWith(".so")) {
```

Comment:
what about the so version symlinks (e.g .so.4)?

Reply:
Changed to `contains`.

Reply:
should be ```endsWith(".so")``` and ```contains(".so.")``` to catch all valid cases

Reply:
But `contains(".so")` will work if it `endsWith(".so")` and if `endsWith(".so.2")`. Any case this filter is just to pass if it's folder. Inside folder there's only `.so` files, because we include to jar by filter: https://github.com/openucx/ucx/blob/master/bindings/java/pom.xml.in#L167

Reply:
once you will have file as ```file.so.good.o``` or .a - it will break

Reply:
It won't break, it'll just unpack it to tmp folder. The purpose of this method is to iterate over content of ucx folder inside jar and for each file - unpack it to tmp folder. The filter is just to bypass `ucx` folder itself